examples/apps/table-detail-tea/src/main.rs

```diff
@@ -0,0 +1,239 @@
+/// reading. See the [`latest`] branch for the code which works with the most recent Ratatui
+/// release.
+///
+/// [`latest`]: https://github.com/ratatui/ratatui/tree/latest
+use std::time::Duration;
+
+use crossterm::event::{
+    self, DisableMouseCapture, EnableMouseCapture, Event, KeyCode, MouseEventKind,
+};
+use crossterm::ExecutableCommand;
+use ratatui::layout::{Constraint, Layout};
+use ratatui::style::{Modifier, Style};
+use ratatui::text::Line;
+use ratatui::widgets::{Block, Borders, Padding, Paragraph, Row, Table, TableState};
+use ratatui::Frame;
+
+static INFO_TEXT: &str = " (Esc/q) quit | (↑) move up | (↓) move down ";
+
+/// Application data model and state
+#[derive(Debug, Default)]
+struct Model {
+    table_items: Vec<Data>,
+    table_state: TableState,
+    running_state: RunningState,
+}
+
+#[derive(Debug, Default, PartialEq, Eq)]
+enum RunningState {
+    #[default]
+    Running,
+    Done,
+}
+
+/// Data model
+#[derive(Debug, Default)]
+struct Data {
+    name: String,
+    address: String,
+    email: String,
+}
+
+#[derive(PartialEq, Copy, Clone)]
+enum Message {
+    SelectNext,
+    SelectPrevious,
+    Quit,
+}
+
+fn main() -> color_eyre::Result<()> {
+    tui::install_panic_hook();
+    let mut terminal = tui::init_terminal()?;
+    let mut model = Model {
+        table_items: generate_some_fans(),
+        ..Default::default()
+    };
+    stdout().execute(EnableMouseCapture)?;
+    while model.running_state != RunningState::Done {
+        // Render the current view
+        terminal.draw(|f| view(&mut model, f))?;
+
+        // Handle events and map to a Message
+        let mut current_msg = handle_event(&model)?;
+
+        // Process updates as long as they return a non-None message
+        while current_msg.is_some() {
+            current_msg = update(&mut model, current_msg.unwrap());
+        }
+    }
+    stdout().execute(DisableMouseCapture)?;
+    tui::restore_terminal()?;
+    Ok(())
+}
+
+fn view(model: &mut Model, frame: &mut Frame) {
+    let [top, bottom] = Layout::vertical([Constraint::Fill(1); 2]).areas(frame.area());
+
+    // Table setup
+    let header = Row::new(vec!["Name", "Address", "Email"])
+        .style(Style::default().add_modifier(Modifier::BOLD))
+        .height(1);
+
+    let rows = model.table_items.iter().map(|data| {
+        Row::new(vec![
+            data.name.as_str(),
+            data.address.as_str(),
+            data.email.as_str(),
+        ])
+        .height(2)
+    });
+
+    let table = Table::new(
+        rows,
+        [
+            Constraint::Length(20),
+            Constraint::Min(10),
+            Constraint::Fill(1),
+        ],
+    )
+    .header(header)
+    .row_highlight_style(Style::default().add_modifier(Modifier::REVERSED | Modifier::ITALIC))
+    .block(
+        Block::default()
+            .borders(Borders::ALL)
+            .title(" Terminal fans "),
+    );
```

Comment:
nit: you can often save a bunch of vertical height due to automatic wrapping by adding an intermediate variables (e.g. widths here and using Block::bordered()) It's not much, but in example code lines of code are often worth saving to make it feel like less code to have to worry about.

```suggestion
        let widths = [
            Constraint::Length(20),
            Constraint::Min(10),
            Constraint::Fill(1),
        ];
        let table = Table::new(rows, widths)
            .header(header)
            .row_highlight_style(Style::new().add_modifier(Modifier::REVERSED | Modifier::ITALIC))
            .block(Block::bordered().title(" Terminal fans "));
```